dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -979,12 +981,64 @@ public static int runCmd(String[] cmd, File workingDirectory, File outputFile) t
         pb.redirectErrorStream(true);
 
         Log.info(c, "runCmd", "Running command " + Arrays.asList(cmd));
-        Process p = pb.start();
-        int exitCode = p.waitFor();
-        return exitCode;
+        
+        int hardTimeout;
+        
+        try {
+            hardTimeout = Integer.parseInt(System.getProperty("fat.timeout"));
+        }catch (NumberFormatException e) {
+            hardTimeout = 10800000;  // 10800000ms = 3hr
+        }
+        
+        long softTimeout = -1;
 
+        // We need to ensure that the hard timeout is large enough to avoid future issues
+        if (hardTimeout >= 30000) {
+            softTimeout = hardTimeout - 15000; // Soft timeout is 15 seconds less than hard timeout
+        }
+
+        Process p = pb.start();
+        int returnCode = 1;
+        boolean returnStatus;
+        if (softTimeout > -1) {
+            returnStatus = p.waitFor(softTimeout, TimeUnit.MILLISECONDS);  // Requires Java 8+
+            if (returnStatus == false) {
+                // Parse through the MVN logs for potential networking issues
+                if (outputFile.exists() && outputFile.canRead()) {
+                    try (Scanner s = new Scanner(outputFile)) {
+                        // Get the last two lines from the MVN log
+                        String lastLine = "";
+                        String secondLastLine = ""; 
+                        while (s.hasNextLine()) {
+                            secondLastLine = lastLine;
+                            lastLine = s.nextLine().toLowerCase();
+                        }
+                        // Check if the last or second line has the text "downloading" or "downloaded"
+                        // Throw custom timeout error rather then the one provided by the JUnitTask
+                        if ((lastLine.contains("downloaded") || lastLine.contains("downloading")) || (secondLastLine.contains("downloaded") || secondLastLine.contains("downloading"))) {       
+                            String timeoutMsg = "Timeout occurred. FAT timeout set to: " + hardTimeout + 
+                                                ". It appears there were some issues gathering dependencies. This may be due to network issues such as slow download speeds.";
+                            Log.info(c, "runCmd", timeoutMsg);  // Log the timeout message into messages.log or the default log 
+                            throw new AssertionFailedError(timeoutMsg);
+                        }
```

Comment:
Lets just include the last 5-10 lines of the output file no matter what, instead of special casing it for the last lines containing certain words.

Reply:
Last 7 line will be included in all cases. Special cases will be left in.